src/exts/discord_integration.js

```diff
@@ -0,0 +1,94 @@
+const { bridge } = require('../spring_api');
+const { log } = require('../spring_log');
+const { config } = require('../launcher_config');
+const DiscordRPC = require("discord-rpc");
+
+if (config) {
+    if (config.discord_rich_presence.application_id == null) {
+        log.warn("config.discord_rich_presence.application_id not defined");
+        return
+    }
+}
+
+// Application ID from https://discord.com/developers/applications/<applicationId>/
+const applicationId = "1185990483143569448";
+const rpc = new DiscordRPC.Client({ transport: 'ipc' });
+
+DiscordRPC.register(applicationId);
+
+bridge.on('DiscordSetActivity', async command => {
+    // command
+    // {
+    //     state - string
+    //     details - string
+    //     startTimestamp - unix timestamp
+    //     playerCount - int > 0
+    //     maxPlayerCount - int > 0
+    //     partyId - string
+    //     largeImageText - string
+    //     smallImageKey - string
+    //     smallImageText - string
+    // }
+
+    if (!rpc) {
```

Comment:
when can this happen when constructor is called above?

Reply:
Never :) I reworked that part